tezos-encoding/src/nom.rs

```diff
@@ -572,6 +573,56 @@ where
     }
 }
 
+#[derive(Debug, PartialEq)]
+/// Lazy, memoized deserialization of a dynamic block.
+pub struct LazyCell<'a, T>
```

Comment:
Should this instead be an enum, to be equivalent to the `lazy` in `Data_encoding`?

```
Bytes(&'a [u8]), 
Value(T),
Both(&'a [u8], T)
```
This might allow slightly easier pattern matching as below

Reply:
I guess the difference is that Data_encoding not only features lazy deserialization but also lazy serialization. Do we need lazy serialization?

I'll try to see if turning the option into an enum simplifies the implementation.

Reply:
Ah good point; probably don't need `lazy_serialization` to be fair